shared/fs/header.js

```diff
@@ -0,0 +1,86 @@
+// @flow
+import * as React from 'react'
+import {globalStyles, globalColors, globalMargins, isMobile} from '../styles'
+import {Avatar, Box, ClickableBox, Icon, Text} from '../common-adapters'
+import HeaderConnector from './header-container'
+
+type FolderHeaderProps = {
+  items: Array<{
+    name: string,
+    path: string,
+  }>,
+  isTeamPath: boolean,
+  onOpenBreadcrumb: (path: string) => void,
+}
+
+const FolderHeader = HeaderConnector(({items, isTeamPath, onOpenBreadcrumb}: FolderHeaderProps) => (
+  <Box>
+    {items.length === 1 ? (
+      <Box style={folderHeaderStyleRoot}>
+        <Text type="BodyBig">Keybase Files</Text>
+      </Box>
+    ) : (
+      <Box style={folderHeaderStyleTree}>
+        {items.length < 4 &&
+          items.map((i, idx) => (
+            <Box key={i.path} style={folderBreadcrumbStyle}>
+              {idx !== 0 && <Icon type="iconfont-back" style={iconStyle} />}
+              {idx === 2 &&
+                isTeamPath && <Avatar size={12} teamname={i.name} isTeam={true} style={styleTeamAvatar} />}
+              {idx === items.length - 1 ? (
+                <Text type="BodyBig">{i.name}</Text>
+              ) : (
+                <ClickableBox onClick={() => onOpenBreadcrumb(i.path)}>
```

Comment:
@chrisnojima: is there risk here that we'll be re-rendering more often than we should?

Reply:
Isn't it instantiating a new closure on each render that causes the equality test to fail?

Reply:
if you're really worried about this (i'm not as this list won't be mutating after mount you can make this a class and point to a method instead
_onOpenBreadcrumb = (p: Path) => this.props.onOpenBreadcrumb(p)